src/contexts/TotalStaked/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Signer } from 'ethers';
+
+import { TOKEN_GEYSER_ADDRESS } from 'constants/index';
+
+import { Web3Context } from 'contexts/Web3';
+
+import { getTokenGeyserContract } from 'utils';
+
+export interface ContextValues {
+  // eslint-disable-next-line
+  totalStaked?: any;
+}
+
+export const TotalStakedContext = React.createContext<ContextValues>({});
+
+export const TotalStakedProvider: React.FC = ({ children }) => {
+  const { address, signer } = React.useContext(Web3Context);
+  const [totalStaked, setTotalStaked] = React.useState<string>();
+
+  const fetchTotalStaked = React.useCallback(async (userAddress: string, signer: Signer) => {
+    const geyserContract = getTokenGeyserContract(signer, TOKEN_GEYSER_ADDRESS as string);
+    try {
+      const balance = await geyserContract.totalStakedFor(userAddress);
+      setTotalStaked(balance.toString());
+    } catch (e) {
+      console.error(e);
+      setTotalStaked('0');
+    }
+  }, []);
+
+  React.useEffect((): (() => void) => {
+    if (address && signer) {
+      fetchTotalStaked(address, signer);
+      const refreshInterval = setInterval(() => fetchTotalStaked(address, signer), 10000);
```

Comment:
Let's move the `10000` to a variable on top of the file and mark it as a constant by writing it in uppercase—something like `RETRY_DELAY`